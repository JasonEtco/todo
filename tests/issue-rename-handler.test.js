const { Application } = require('probot')
const issueEdited = require('./fixtures/payloads/issues.edited.json')
const plugin = require('..')

describe('issue-rename-handler', () => {
  let app, github, event

  beforeEach(() => {
    app = new Application()
    event = { name: 'issues', payload: issueEdited }

    github = {
      issues: {
        update: jest.fn(),
        createComment: jest.fn()
      }
    }

    app.auth = jest.fn(() => Promise.resolve(github))
    app.load(plugin)
  })

  it('un-edits the issue title', async () => {
    await app.receive(event)
    expect(github.issues.update.mock.calls[0][0]).toMatchSnapshot()
    expect(github.issues.createComment.mock.calls[0][0]).toMatchSnapshot()
  })

  it('only acts if the title is edited', async () => {
    event.payload.changes = {}
    await app.receive(event)
    expect(github.issues.update).not.toHaveBeenCalled()
    expect(github.issues.createComment).not.toHaveBeenCalled()
  })
})
