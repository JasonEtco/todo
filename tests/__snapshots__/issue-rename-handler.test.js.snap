// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue-rename-handler un-edits the issue title 1`] = `
Object {
  "title": "I have been edited",
}
`;

exports[`issue-rename-handler un-edits the issue title 2`] = `
Object {
  "body": "Please do not change the issue title! **todo** uses it to prevent duplicate issues from being opened. If you think this is an error, please [open an issue](https://github.com/jasonetco/todo)!",
}
`;
